wunderio/elasticsearch_helper
# Feature: Unique queue items

### Feature

This changes makes the Elasticsearch helper queue to only index unique items to prevent redundant items in the queue table.

Changes:
- Add a custom Queue and QueueFactory for Elasticsearch, these extends the core DatabaseQueue and QueueDatabaseFactory.
- Decorate the queue service so that the QueueFactory could be injected without hardcoding it in settings.php (see https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Queue%21QueueFactory.php/function/QueueFactory%3A%3Aget/8.8.x)

### Testing
1. Create a few entities for indexing
2. Execute `drush eshr`
3. Inspect and verify that the indexing queue is populated with entities for indexing
4. Execute `drush eshr` again
5. Inspect the verify that the indexing queue does not contain duplicate items
6. Run `drush cron` to index the queue items
7. Verify that indexing works
